avssync.go

```diff
@@ -48,16 +48,16 @@ func NewAvsSync(
 		syncInterval:                 syncInterval,
 		operators:                    operators,
 		quorums:                      quorums,
-		fetchQuorumsDynamically:      fetchQuorumsDynamically,
+		noFetchQuorumsDynamically:    noFetchQuorumsDynamically,
 		retrySyncNTimes:              retrySyncNTimes,
 		readerTimeoutDuration:        readerTimeoutDuration,
 		writerTimeoutDuration:        writerTimeoutDuration,
 	}
 }
 
 func (a *AvsSync) Start() {
-	a.logger.Infof("Starting avs sync with sleepBeforeFirstSyncDuration=%s, syncInterval=%s, operators=%v, quorums=%v, fetchQuorumsDynamically=%v, readerTimeoutDuration=%s, writerTimeoutDuration=%s",
-		a.sleepBeforeFirstSyncDuration, a.syncInterval, a.operators, a.quorums, a.fetchQuorumsDynamically, a.readerTimeoutDuration, a.writerTimeoutDuration)
+	a.logger.Infof("Starting avs sync with \n  sleepBeforeFirstSyncDuration=%s, \n  syncInterval=%s,\n  operators=%v,\n  quorums=%v,\n  noFetchQuorumsDynamically=%v,\n  readerTimeoutDuration=%s,\n  writerTimeoutDuration=%s",
```

Comment:
having new lines in a log line can complicate loading the log to a central location (ELK)

Reply:
Any ideas on how to make it both easy to pipe to a central logging system, but also make it human readable for users who just want to look at the terminal?

I found it hard to read when dumping the structured json log. In inc-sq, I was piping output through https://github.com/maoueh/zap-pretty, but I don't want to make the assumption that users are using zap logger.

Reply:
What worked for me in previous companies when the binary executed within a docker was to have two different hooks for the logger. One that writes in human-readable usually goes to a local file. The other went to the docker stdout in a JSON format that will be easier for parsing in the central login system.
The local file was persistent, so if the docker container was recreated, the file remains. Standard loggers in Golang can manage the number of files and the size limit of each file.

I also implemented a simple non-locking logger, which gets the message in an unbuffered channel and processes the log hooks on a different go routine (so the core code was not blocked).

If you want to allow others to use your module with their logging system, then you should create an interface for the logging and enable the external module to call you with their implementation of the logger. Then, it will be the external client that implements the logger hooks.

Reply:
Ok removed newlines in [0ce1431](https://github.com/Layr-Labs/avs-sync/pull/6/commits/0ce143169eb94c7d0751faa8135a8715f168e2c7)

Re writing to two separate files, that's a neat idea. If you don't mind I created a new [issue](https://github.com/Layr-Labs/avs-sync/issues/10) so I can merge this PR in and will get to the logger later (prob not super soon though..)

For the files though, wouldn't you want the opposite way? That machine-readable json should go to a permanent file (afaik logging systems like promtail read from files and send them over to the centralized logging system like grafana loki), and the human-readable should go to docker std output, so a human can just `docker logs ...` and read them?

Re importing your own logger, that is totally possible! You just have to fork avs-sync, the sdk uses a [logger interface](https://github.com/Layr-Labs/eigensdk-go/blob/master/logging/logger.go) so you could totally come with your own logger.